packages/celotool/src/e2e-tests/governance_tests.ts

```diff
@@ -687,6 +687,15 @@ describe('governance tests', () => {
         }
       }
     })
+
+    it('should have emitted transfer events when paying epoch rewards', async () => {
```

Comment:
Can we test that the correct events are emitted, and also that the events are only emitted during epoch blocks?

Reply:
Ok, adding more checks.

Reply:
Ah, didn't realize that we don't actually emit the main events that we'd want to record.

More specifically, can we check that the following events are emitted in finalizing the last block of the epoch, and that no events are emitted in finalizing any other block?

`TargetVotingYieldUpdated` (needs to be added to `EpochRewards.updateTargetVotingYield`)
`ValidatorEpochPaymentDistributed` (needs to be added to `Validators.distributeEpochPayementsFromSigner`)
`EpochRewardsDistributedToVoters` (needs to be added to `Election.distributeEpochRewards`)

And can you please add unit tests that check for these events as well?

Thanks!